main.tf

```
@@ -4,6 +4,12 @@ resource "github_membership" "maintainers" {
   role     = "member"
 }
 
+resource "github_team_membership" "maintainers" {
+  for_each = toset(var.maintainers)
```

```suggestion
  for_each = toset(compact(distinct(concat(var.maintainers, var.board))))
```
The board and maintainers are separate lists but both should be in this group so they get notified on PRs etc.